.scripts/ci.yml

```diff
@@ -1,35 +1,117 @@
 trigger:
   - v6
 
+parameters:
+  - name: PublishAlpha
+    type: boolean
+    default: false
+  - name: LinuxPool
+    type: string
+    default: azsdk-pool-mms-ubuntu-1804-general
+  - name: IncludeRelease
+    type: boolean
+    default: true
+
 pool:
   vmImage: "ubuntu-latest"
 
-jobs:
-  - job: Build
-    steps:
-      - template: build.yml
-      - script: npm run test
-        displayName: "Run Tests"
-
-      - script: npm run check:tree
-        displayName: "Check git Tree"
-
-      - script: |
-          npm run coverage-push -- $(Build.Repository.Name) $(Build.SourceBranch) $(github-token) $(storage-coverage-user) $(storage-coverage-pass)
-        workingDirectory: node_modules/@microsoft.azure/autorest.testserver
-        displayName: "Upload Coverage Report"
-
-  - template: smoke-test.yml
-    parameters:
-      name: SmokeTest_1
-      tag: ci_1
-
-  - template: smoke-test.yml
-    parameters:
-      name: SmokeTest_2
-      tag: ci_2
-
-  - template: smoke-test.yml
-    parameters:
-      name: SmokeTest_3
-      tag: ci_3
+stages:
+  - stage: Build
+    jobs:
+      - job: Build
+        steps:
+          - template: build.yml
+          - script: npm pack
+          - task: CopyFiles@2
+            inputs:
+              contents: "*.tgz"
+              targetFolder: $(Build.ArtifactStagingDirectory)
+              flattenFolders: true
+            displayName: "Copy packages"
+
+          - task: PublishPipelineArtifact@1
+            condition: succeededOrFailed()
+            displayName: "Publish artifacts"
+            inputs:
+              artifactName: packages
+              path: $(Build.ArtifactStagingDirectory)
+
+      - job: Test
```

Comment:
This step should also depend on build. right? There is no point in running them if build fails.